### DISCUSSION TITLE
How to reduce wrong gender issues?

### DISCUSSION THREAD

**Discussion by qgustavor** [machinewrapped/llm-subtrans#197] [Q&A] [answered]
I tried translating using Google Gemini to translate a 1995 TV series and, while it did better than using Google Translate on Subtitle Edit, it made a lot of gender-related mistakes, like mistaking "she" for "he". Considering it's an old TV series and I mentioned it, I guessed it could probably could infer gender from its training data, but the results showed otherwise. Do you know any way to improve that? Maybe some change in the instructions, I guess.

**Comment by machinewrapped** [comment, marked as answer]
That's a tricky one - what was the original language? 

Which Gemini model were you using? Flash is a little bit pea-brained, so maybe try it with Gemini 1.5 pro or gpt4o if you were using that.

You can add information to the project description indicating the gender of specific characters, especially if it's not clear from the name. You could try adding genders to names in the Names list as well - never tried it, but the smarter models like gpt4o might be able to use the information.

![image](https://github.com/user-attachments/assets/063b7811-18f7-484d-a151-dbb8882bcaa9)

**Reply by qgustavor** [reply]
English to Portuguese, since my wife doesn't know English and I won't want to get spoiled by translating it myself. I tried the flash model since I wanted quick results and assumed it would be fine. I will try other models. I will also try specifying genders.

I wrote my question in a generic way to avoid getting in language specifics. Like, Portuguese has a lot of gendered words which aren't gendered in English, so talking about specifics in English isn't easy.

The most common mistake was using feminine they when there was a man and a woman in the scene, so it should use masculine they. That said, I guess that even for the more advanced models, that's hard to guess from the script alone, as it hardly has any info about who were in that scene, unless they were trained with some specific info about the show I was translating or the book it's based on.

That said, that's not an issue exclusive to English to Portuguese translations, it also happens with Japanese to English: 田中さん could be either Mr. Tanaka or Mrs. Tanaka depending on context. Some translators (mostly fan translators) avoid this issue by not translating さん, transliterating to -san instead. In the case of English to Portuguese translations probably the closest trick is to use the masculine forms of words in case the LLM can't infer the gender, as it's the "default" gender in Portuguese and, even in real life, many woman use masculine words.

I remember reading a paper on how to improve machine translations on how to improve translations that depend on multimodal context, but I guess that's out of the scope of this project. I'm already happy with the help I got here.

**Comment by machinewrapped** [comment]
Yeah, unfortunately mixing up the speaker is a hard nut to crack until the model can actually watch the film/show and see the full context for the line, but the smarter models are definitely better at inferring context. Sometimes this can lead them to make bigger mistakes though, where they get the wrong idea about a line and carry it forward!

I suppose closed captions that include speaker annotations would be better for translation, but comparatively harder to come by.

Since you're translating multiple episodes of the same show to the same language it may be worth creating custom instructions, where you can give specific guidance and examples. You can save them to a file and set them as the default in the app settings until you've finished the show.

![image](https://github.com/user-attachments/assets/f8567b47-8e1e-46d3-8bd2-9e88ac8f7ec2)
![image](https://github.com/user-attachments/assets/2dc5869b-e00f-4d48-a315-6a931cd6764b)
